### ISSUE TITLE
Weird Algorithm

### ISSUE THREAD

**Issue by harlankoehn** [DanielSantoro/wc-password-strength-settings#1]
The following passwords are considered weak:

Google.123!
Google#123!
Goo123gl!
Harlan;670!

This seems like a bug or are these passwords truly considered insecure?

NOTE, I've not exposed any passwords I actually use for anything.

**Comment by AthenaTzeng** [comment]
The strength check algo is NOT a part of this plugin. (it is in wordpress core)
This plugin change only the "threshold" of the min strength while checking
You can try to trace from Line22 in wp-admin/js/password-strength-meter.js (I use wordpress version 4.6.1)

**Comment by DanielSantoro** [comment]
Sorry for the delay in getting back to you here - my GitHub notifications seem to have been not working. 

The algorithm used is the same one WordPress uses (thanks for mentioning that, @AthenaTzeng !), which is built from [zxcvbn developed by Dropbox](https://github.com/dropbox/zxcvbn). I don't know specifically what determines a strong password vs. a weak one, but they do have a page where you can test: https://dl.dropboxusercontent.com/u/209/zxcvbn/test/index.html

In some cases it's saying something is relatively safe, such as Google#123!, but the issue is that Google is in the dictionary (and is easily brute forced), and `#123!` is also pretty commonly bypassed by machines. This isn't to say that's the weakest password out there, but that it can be stronger. The best passwords I've found are three or four short dictionary words with spaces and special characters if you need it - you can make a quick one by looking around. In my case, I have a soda can near my desk, so I could use something like:

`Can coaster desk 3` - I got the 3 from the number of words in the soda, Diet Mountain Dew. So, if I just remember that image then I can remember the password which is often more secure than entirely random strings of letters and numbers.

Long story short, password strength meters are strange. :)
